language/tools/move-package/src/resolution/dependency_graph.rs

```diff
@@ -904,3 +946,41 @@ fn str_escape(s: &str) -> Result<String, fmt::Error> {
 fn path_escape(p: &Path) -> Result<String, fmt::Error> {
     str_escape(p.to_str().ok_or(fmt::Error)?)
 }
+
+fn format_deps(msg: &str, deps: Vec<String>) -> String {
+    let mut s = "".to_string();
+    if !deps.is_empty() {
+        s.push_str(msg);
+        for d in deps {
+            s.push_str("\n\t");
+            s.push_str(&d);
+        }
+    }
+    s
+}
+
+/// Checks if dependencies of a given package in two different dependency graph maps are the
+/// same.
+fn pkg_deps_equal(
+    pkg_name: Symbol,
+    pkg_graph: &DiGraphMap<PM::PackageName, Dependency>,
+    other_graph: &DiGraphMap<PM::PackageName, Dependency>,
+) -> (Vec<String>, Vec<String>) {
+    let pkg_edges = BTreeSet::from_iter(pkg_graph.edges(pkg_name).map(|(_, pkg, dep)| (dep, pkg)));
+    let other_edges =
+        BTreeSet::from_iter(other_graph.edges(pkg_name).map(|(_, pkg, dep)| (dep, pkg)));
+
+    let (pkg_deps, other_deps): (Vec<_>, Vec<_>) = pkg_edges
+        .symmetric_difference(&other_edges)
+        .partition(|dep| pkg_edges.contains(dep));
+    (
+        pkg_deps
+            .iter()
+            .map(|(dep, pkg)| format!("{}", DependencyTOML(*pkg, dep)))
```

Comment:
nit: Seeing as you have a separate function for formatting, I would maybe put the logic of calling `format!` into there.

Reply:
My Rust mojo definitely still has gaps and I could not easily return `Dependency` from a function, presumably because it contains (optional) `BTreeMap`s...

Reply:
Never mind, I think I figure it out.

Reply:
It's possible that you could get into issues with temporaries if returning by value, but I think returning by reference is the right call here, in any case!